bigtable-hbase-1.x-parent/bigtable-hbase-1.x-mapreduce/src/main/java/com/google/cloud/bigtable/mapreduce/validation/BigtableSyncTableJob.java

```diff
@@ -265,34 +279,69 @@ public boolean verifyRequiredArgsSet(
       String sourceBigtableInstance,
       String targetBigtableProjectId,
       String targetBigtableInstance) {
+
+    // neither source zk or source bigtable config set
     if (BigtableSyncTableAccessor.getSourceZkCluster(syncTable) == null
-        && (sourceBigtableProjectId == null || sourceBigtableInstance == null)) {
+        && (sourceBigtableProjectId == null && sourceBigtableInstance == null)) {
       printUsage(
           "--sourcezkcluster or --sourcebigtableproject and --sourcebigtableinstance required.",
           args);
       return false;
     }
 
+    // both source zk and source bigtable configs are set
+    if (BigtableSyncTableAccessor.getSourceZkCluster(syncTable) != null
+        && (sourceBigtableProjectId != null || sourceBigtableInstance != null)) {
+      printUsage(
+          "--sourcezkcluster and --sourcebigtableproject and --sourcebigtableinstance are set. Set one or the other.",
+          args);
+      return false;
+    }
+
+    // source bigtable project or source instance not set
+    if (BigtableSyncTableAccessor.getSourceZkCluster(syncTable) == null
+        && ((sourceBigtableProjectId == null || sourceBigtableInstance == null))) {
+      printUsage("--sourcebigtableproject and --sourcebigtableinstance required.", args);
+      return false;
+    }
+
+    // neither target zk or target bigtable config set
     if (BigtableSyncTableAccessor.getTargetZkCluster(syncTable) == null
-        && (targetBigtableProjectId == null || targetBigtableInstance == null)) {
```

Comment:
I think this was correct. if any of instance/project is null, its basically malformed.

Reply:
i perform 2 separate validations for hbase or bigtable connection configurations to provide better usage instructions. 

First check if one or the other is set (zk or bigtable) and if neither are set, print usage that zk or bigtable configs are required. 

Later, check if zk is not set, and check bigtable project/instance configs are both set. if not, we print usage of all bigtable configs that are are required.